tests/shared.py

```diff
@@ -30,6 +32,51 @@ def new_backend():
     return Backend(host, port)
 
 
+def driver_feature(feature):
+    if not isinstance(feature, Feature):
+        raise Exception('The argument should be instance of Feature')
+
+    def get_valid_test_case(*args, **kwargs):
+        if not args or not isinstance(args[0], TestkitTestCase):
+            raise Exception('Should only decorate TestkitTestCase methods')
+        return args[0]
+
+    def driver_feature_decorator(func):
+        def wrapper(*args, **kwargs):
+            test_case = get_valid_test_case(*args, **kwargs)
+            if (feature.value not in test_case._driver_features):
+                test_case.skipTest("Needs support for %s" % feature.value)
+            return func(*args, **kwargs)
+        return wrapper
+    return driver_feature_decorator
+
+
+features = None
+
+
+def get_driver_features(backend):
+    global features
+    if features is None:
+        # TODO Remove when dotnet implements the GetFeature message
+        if get_driver_name() in ['dotnet']:
+            warnings.warn("""Driver does not implements GetFeatures.
+            """)
```

Comment:
Is there a good reason for this to be a multi-line sting? Especially given the other warning is a regular string?

Reply:
It was a leftover. I will fix it.